src/Microsoft.AspNet.OData.Shared/DefaultContainerBuilder.cs

```diff
@@ -20,9 +21,9 @@ public class DefaultContainerBuilder : IContainerBuilder
         /// <param name="lifetime">The lifetime of the service to register.</param>
         /// <param name="serviceType">The type of the service to register.</param>
         /// <param name="implementationType">The implementation type of the service.</param>
-        /// <returns>The <see cref="IContainerBuilder"/> instance itself.</returns>
-        public virtual IContainerBuilder AddService(
-            ServiceLifetime lifetime,
+        /// <returns>The <see cref="Microsoft.OData.IContainerBuilder"/> instance itself.</returns>
+        public virtual Microsoft.OData.IContainerBuilder AddService(
```

Comment:
> Microsoft.OData. [](start = 22, length = 17)

so do you still need these?

Reply:
Not the ones on IContainerBuilder but the others are needed to disambiguate OData vs DependencyInjection.

---
In reply to: [139848023](https://github.com/OData/WebApi/pull/1075#discussion_r139848023) [](ancestors = 139848023)